tools/generators/epic-generator/index.ts

```diff
@@ -0,0 +1,191 @@
+    throw new Error('You are not logged into GitHub CLI (gh).');
+  }
+};
+
+interface Package {
+  name: string;
+  folder: string;
+  owners: string[];
+}
+
+interface Ownership {
+  path: string;
+  owners: string[];
+}
+
+interface MigrationIssue {
+  issueUrl?: string;
+  assignee?: string;
+  packages: Package[];
+}
+
+type MigrationIssues = Record<string, MigrationIssue>;
+
+const getConvergedPackages = (tree: Tree) => {
+  const projects = getProjects(tree);
+
+  const convergedPackages: Package[] = [];
+  projects.forEach((project, key) => {
+    if (project.projectType === 'library' && isPackageConverged(tree, project)) {
+      convergedPackages.push({
+        name: key,
+        folder: project.root,
+        owners: [],
+      });
+    }
+  });
+
+  return convergedPackages;
+};
+
+const getCodeowners = (): Ownership[] => {
+  const codeownersContent = readFileSync('.github/CODEOWNERS', 'utf8');
+
+  return codeownersContent.split('\n').map(line => {
+    const [path, ...owners] = line.split(' ');
+
+    return {
+      path: path.replace(/\/+$/, ''),
+      owners,
+    };
+  });
+};
+
+const mapOwnerships = (packages: Package[], ownerships: Ownership[]): Package[] => {
+  return packages.map(pckg => {
+    const owners = ownerships.find(ownership => ownership.path === pckg.folder)?.owners || [];
+
+    return {
+      ...pckg,
+      owners,
+    };
+  });
+};
+
+const getPackages = (tree: Tree) => {
+  const packages = getConvergedPackages(tree);
+
+  const ownerships = getCodeowners();
+
+  return mapOwnerships(packages, ownerships);
+};
+
+const createEpic = (repo: string, title: string, message: string) => {
+  const issueUrl = execSync(`gh issue create --repo "${repo}" --title "${title}" --body "${message}"`)
+    .toString()
+    .trim();
+
+  return issueUrl;
+};
+
+const createIssue = (repo: string, issue: MigrationIssue, templateTitle: string) => {
+  const title = `${templateTitle} - ${issue.assignee || 'ownerless'}`;
+  const message = stripIndents`
+    🚧 This is an auto-generated issue to individually track migration progress.
+
+    ### Packages to migrate:
+    ${issue.packages.map(pckg => `- ${pckg.name}`).join('\n')}
+  `;
+
+  const command = `gh issue create --repo "${repo}" --title "${title}" --body "${message}"`;
+
+  const issueUrl = execSync(command).toString().trim();
+
+  return issueUrl;
+};
+
+const generateIssues = (repo: string, templateTitle: string, packages: Package[]) => {
+  const migrationIssues = packages.reduce<MigrationIssues>((acc, pckg) => {
+    const teamOwner = pckg.owners.find(owner => owner.startsWith('@microsoft/'));
+    const key = teamOwner || 'ownerless';
+
+    if (key in acc) {
+      acc[key].packages.push(pckg);
+    } else {
+      acc[key] = {
+        assignee: teamOwner,
+        packages: [pckg],
+      };
+    }
+
+    return acc;
+  }, {});
+
+  Object.entries(migrationIssues).forEach(([key, issue]) => {
```

Comment:
wondering if this additional iteration is needed. I checked that createIssue is not using issueUrl thus this can be setup within original reduce.

```ts
if (acc[key]) {
      acc[key].packages.push(pckg);
      return acc;
    }

    const issue = {
      assignee: teamOwner,
      packages: [pckg],
    };
    

    acc[key] = {
      ...issue,
      issueUrl: createIssue(repo, issue, templateTitle),
    };

    return acc;
```

Reply:
That can't be used as all packages need to be accounted for to create the issue. If you check `createIssue`, you'll see that the message lists all the team's ownerships in it:

```
const message = stripIndents`
  🚧 This is an auto-generated issue to individually track migration progress.

  ### Packages to migrate:
  ${issue.packages.map(pkg => `- ${pkg.name}`).join('\n')}
`;
```

By leveraging your suggestion, issues would be created only with the first package ownership found.